text/3537-msrv-resolver.md

```diff
@@ -0,0 +1,1238 @@
+particularly when no MSRV is specified,
+but this shouldn't affect existing `Cargo.lock` files since the currently
+resolved dependencies always get preference.
+
+This can be overridden with `--ignore-rust-version` and config's `resolver.precedence`.
+
+Implications
+- If you use `cargo update --precise <msrv-incompatible-ver>`, it will work
+- If you use `--ignore-rust-version` once, you don't need to specify it again to keep those dependencies though you might need it again on the next edit of `Cargo.toml` or `cargo update` run
+- If a dependency doesn't specify `package.rust-version` but its transitive dependencies specify an incompatible `package.rust-version`,
+  we won't backtrack to older versions of the dependency to find one with a MSRV-compatible transitive dependency.
+- A package with multiple MSRVs, depending on the features selected, can still do this as version requirements can still require versions newer than the MSRV and `Cargo.lock` can depend on those as well.
+
+As there is no `workspace.rust-version`,
+the resolver will pick the lowest version among workspace members.
+This will be less optimal for workspaces with multiple MSRVs and dependencies unique to the higher-MSRV packages.
+Users can workaround this by raising the version requirement or using `cargo update --precise`.
+
+When `rust-version` is unset,
+we'll fallback to `rustc --version`.
+This is primarily targeted at helping users with a
+[`rust-toolchain.toml` file](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file)
+(to reduce duplication)
+though this would also help users who happen to be on an old rustc, for whatever reason.
+As this is just a preference for resolving dependencies, rather than prescriptive,
+this shouldn't cause churn of the `Cargo.lock` file.
+We already call `rustc` for feature resolution, so hopefully this won't have a performance impact.
+
+## Cargo config
+
+We'll add a `resolver.precedence ` field to `.cargo/config.toml` which will control the package version prioritization policy.
+
+```toml
+[build]
+resolver.precedence = "rust-version"  # Default with `v3`
+```
+with potential values being:
+- `maximum`: behavior today (default for v1 and v2 resolvers)
+  - Needed for [verifying latest dependencies](https://doc.rust-lang.org/nightly/cargo/guide/continuous-integration.html#verifying-latest-dependencies)
+- `minimum` (unstable): `-Zminimal-versions`
+  - As this is just precedence, `-Zdirect-minimal-versions` doesn't fit into this
+- `rust-version`:  what is defined in the package (default for v3 resolver)
+- `rust-version=` (future possibility)
+  - `package`: long form of `rust-version`
+  - `rustc`: the current running version
+    - Needed for "separate development / publish MSRV" workflow
+  - `<x>[.<y>[.<z>]]`: manually override the version used
+
+If a `rust-version` value is used, we'd switch to `maximum` when `--ignore-rust-version` is set.
```

Comment:
The wording on here is confusing to me. Does this mean that explicitly setting `precedence = maximum` is the same as passing `--ignore-rust-version`?

Reply:
This is more intended as a rational for why we aren't proposing every command get a `--ignore-rust-version` flag.

I've moved this down into the Alternatives and clarified the wording in 
cf4651b  and 13fcb24